source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -434,6 +445,17 @@ ReadOrParseState Filter::readProxyHeader(Network::IoHandle& io_handle) {
     if (nread < 1) {
       ENVOY_LOG(debug, "failed to read proxy protocol (no bytes read)");
       return ReadOrParseState::Error;
+    } else if (nread < PROXY_PROTO_V2_HEADER_LEN &&
```

Comment:
I don't think this handles the case where there is no PROXY header, but the number of bytes read is larger than `PROXY_PROTO_V2_HEADER_LEN`, eg the filter reads an entire HTTP/1 request header.

Reply:
good catch! addressed in https://github.com/envoyproxy/envoy/pull/18951/commits/fa9746a917e6f13ed74eac94c578958e05428e23

i went ahead and wrote a test that failed (the new one), then removed the faulty logic. now there are two tests (for small and large requests that get passed through the filter) that hit the exact same code path. happy to remove either one, what do you think?

Reply:
Please leave both tests in